sdk/appcenter/src/main/java/com/microsoft/appcenter/AppCenter.java

```diff
@@ -442,9 +442,7 @@ private synchronized boolean instanceConfigure(Application application, String a
 
             /* A value with no key is default to the app secret. */
             if (!keyValueSplitter.hasNext()) {
-                if (keyValue.indexOf(KEY_VALUE_DELIMITER) == -1) {
```

Comment:
The reason behind this test was in the case of "appsecret=xxxxxxx;target=", the "target" would become the app secret without the extra check. But I think you are right in a way, no need for multiplying checks like that for corner cases we should instead validate the whole string via regex before parsing it.  If that regex check is really quick to implement for you then go ahead otherwise I'll create a task for it, just let me know.

Reply:
It didn't check anything. This check was fully equal to check above.

Reply:
Yes. I mean `if (!keyValueSplitter.hasNext())`  does the same thing